gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageReadChannel.java

```diff
@@ -1060,6 +1060,42 @@ protected InputStream openStream(long bytesToRead) throws IOException {
     }
   }
 
+  private void cacheFooterWithRetries(Storage.Objects.Get getObject, HttpResponse response)
+      throws IOException {
+    for (int retriesCount = 0; retriesCount < maxRetries; retriesCount++) {
+      try {
+        cacheFooter(response);
+        if (retriesCount != 0) {
+          logger.atInfo().log(
+              "Successfully cached footer after %s retries for '%s'", retriesCount, resourceId);
+        }
+        break;
+      } catch (IOException footerException) {
+        logger.atInfo().withCause(footerException).log(
+            "Failed to prefetch footer (retry #%s/%s) for '%s'",
+            retriesCount + 1, maxRetries, resourceId);
+        if (retriesCount == 0) {
+          readBackOff.get().reset();
+        }
+        if (retriesCount == maxRetries) {
+          resetContentChannel();
```

Comment:
Why we need to reset the Channel? Channel is not even created till this point. I do understand there is no harm in resetting it but it's adding a confusion about channel existence.

IIUC, contentChannel would only exist if there is already a `openStream`.